tooling/nargo_cli/src/cli/test_cmd.rs

```diff
@@ -185,14 +186,47 @@ fn run_test<S: BlackBoxFunctionSolver<FieldElement> + Default>(
 
     let blackbox_solver = S::default();
 
-    nargo::ops::run_test(
-        &blackbox_solver,
-        &mut context,
-        test_function,
-        show_output,
-        foreign_call_resolver_url,
-        compile_options,
-    )
+    let test_function_has_no_arguments = context
+        .def_interner
+        .function_meta(&test_function.get_id())
+        .function_signature()
+        .0
+        .is_empty();
+
+    if test_function_has_no_arguments {
+        nargo::ops::run_test(
+            &blackbox_solver,
+            &mut context,
+            test_function,
+            show_output,
+            foreign_call_resolver_url,
+            compile_options,
+        )
+    } else {
+        use noir_fuzzer::FuzzedExecutor;
+        use proptest::test_runner::TestRunner;
+
+        let compiled_program =
+            compile_no_check(&mut context, compile_options, test_function.get_id(), None, false);
+        match compiled_program {
+            Ok(compiled_program) => {
+                let runner = TestRunner::default();
+
+                let fuzzer = FuzzedExecutor::new(compiled_program.into(), runner);
+
+                let result = fuzzer.fuzz();
+                if result.success {
+                    TestStatus::Pass
+                } else {
+                    TestStatus::Fail {
+                        message: result.reason.unwrap_or_default(),
+                        error_diagnostic: None,
+                    }
+                }
+            }
+            Err(err) => TestStatus::CompileError(err.into()),
+        }
+    }
```

Comment:
We can't integrate this more cleanly into the `run_test` function without https://github.com/noir-lang/noir/pull/5162 as otherwise we get a circular dependency.